control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go

```diff
@@ -92,3 +95,58 @@ func ReconcileDefaultIngressControllerCertSecret(certSecret *corev1.Secret, sour
 	certSecret.Data[corev1.TLSPrivateKeyKey] = sourceSecret.Data[corev1.TLSPrivateKeyKey]
 	return nil
 }
+
+func ReconcileDefaultIngressPassthroughService(service *corev1.Service, defaultNodePort *corev1.Service, hcp *hyperv1.HostedControlPlane) error {
+
+	detectedHTTPSNodePort := int32(0)
+
+	ownerRef := config.OwnerRefFrom(hcp)
+	for _, port := range defaultNodePort.Spec.Ports {
+		if port.Port == 443 {
+			detectedHTTPSNodePort = port.NodePort
+			break
+		}
+	}
+
+	if detectedHTTPSNodePort == 0 {
+		return fmt.Errorf("unable to detect default ingress NodePort https port")
+	}
+
+	service.Spec = corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
```

Comment:
I don't have a good explanation for this yet, but adding port 80 and 443 at the same time to the service breaks the console operators health check. This might be something we need to investigate independently.

Because of this weirdness, I propose that we use an annotation to toggle on the passthrough rather than adding it to the api right now. That way we can back out this functionality later on without breaking API compatiblity if it turns out we can't support 443 and 80 at the same time. I'll make this change.


I did investigate this briefly, and i'm a bit stumped. The console url works fine from my browser but the console health check returns the following error.
`
console   4.12.0-ec.5   False       False         False      13m     RouteHealthAvailable: failed to GET route (https://console-openshift-console.apps.dvossel-hyper1.apps.dvossel-test58.cnv.azure.devcluster.openshift.com): Get "https://console-openshift-console.apps.dvossel-hyper1.apps.dvossel-test58.cnv.azure.devcluster.openshift.com": http: server gave HTTP response to HTTPS client
`

I even made some golang code to test the url and it returns that TLS is enabled.

```
package main

import (
        "crypto/tls"
        "fmt"
        "net/http"
)

func main() {
        client := &http.Client{
                Transport: &http.Transport{
                        TLSClientConfig: &tls.Config{
                                InsecureSkipVerify: true,
                        },
                },
        }
        req, _ := http.NewRequest(http.MethodGet, "https://console-openshift-console.apps.dvossel-hyper1.apps.dvossel-test58.cnv.azure.devcluster.openshift.com:443", nil)
        resp, err := client.Do(req)
        if err != nil {
                panic(err)
        }
        fmt.Printf("resp: %s\n", *resp)
        if resp.TLS != nil {
                fmt.Printf("GOT TLS RESPONSE\n")
        }
}
```

Reply:
Wow, i tested this some more and figured out something even more strange.

If i curl the https url from within the guest cluster, it actually does report that http response to an https request... However, outside of the guest cluster it works fine.